### ISSUE TITLE
Crash caused by navigation code

### ISSUE THREAD

**Issue by Griiimon** [godotengine/godot#67686]
### Godot version

3.5.2rc

### System information

Windows 10

### Issue description

Repeated crashes in my project.

```
CrashHandlerException: Program crashed
Engine version: Godot Engine v3.5.2.rc.custom_build
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
[0] NavMap::get_path (D:\_Godot\Engine source\mygodot-3.5\modules\navigation\nav_map.cpp:179)
[1] NavMap::get_path (D:\_Godot\Engine source\mygodot-3.5\modules\navigation\nav_map.cpp:179)
[2] GodotNavigationServer::map_get_path (D:\_Godot\Engine source\godot-3.5\modules\navigation\godot_navigation_server.cpp:232)
[3] NavigationAgent::update_navigation (D:\_Godot\Engine source\godot-3.5\scene\3d\navigation_agent.cpp:458)
[4] NavigationAgent::get_final_location (D:\_Godot\Engine source\godot-3.5\scene\3d\navigation_agent.cpp:382)
[5] NavigationAgent::is_target_reachable (D:\_Godot\Engine source\godot-3.5\scene\3d\navigation_agent.cpp:372)
[6] MethodBind0R<NavigationAgent,bool>::call (D:\_Godot\Engine source\godot-3.5\core\method_bind.gen.inc:319)
[7] Object::call (D:\_Godot\Engine source\godot-3.5\core\object.cpp:918)
[8] Variant::call_ptr (D:\_Godot\Engine source\godot-3.5\core\variant_call.cpp:1229)
[9] GDScriptFunction::call (D:\_Godot\Engine source\godot-3.5\modules\gdscript\gdscript_function.cpp:1053)
[10] GDScriptInstance::call_multilevel (D:\_Godot\Engine source\godot-3.5\modules\gdscript\gdscript.cpp:1211)
[11] Node::_notification (D:\_Godot\Engine source\godot-3.5\scene\main\node.cpp:58)
[12] Object::notification (D:\_Godot\Engine source\godot-3.5\core\object.cpp:929)
[13] SceneTree::_notify_group_pause (D:\_Godot\Engine source\godot-3.5\scene\main\scene_tree.cpp:1092)
[14] SceneTree::idle (D:\_Godot\Engine source\godot-3.5\scene\main\scene_tree.cpp:608)
[15] Main::iteration (D:\_Godot\Engine source\godot-3.5\main\main.cpp:2298)
[16] OS_Windows::run (D:\_Godot\Engine source\godot-3.5\platform\windows\os_windows.cpp:3543)
[17] widechar_main (D:\_Godot\Engine source\godot-3.5\platform\windows\godot_windows.cpp:176)
[18] _main (D:\_Godot\Engine source\godot-3.5\platform\windows\godot_windows.cpp:200)
[19] main (D:\_Godot\Engine source\godot-3.5\platform\windows\godot_windows.cpp:210)
[20] __scrt_common_main_seh (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[21] BaseThreadInitThunk
-- END OF BACKTRACE --
```

### Steps to reproduce

Can't reproduce consistently, but happend several times. 

I use a custom navigation mesh (via `create_from_mesh()`) on a terrain with objects that is created at runtime. Agents moved as expected on the mesh, so it was working until it crashed between let's say 15 - 90 seconds after starting the game.

**Comment by Calinou** [comment]
@Griiimon Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by smix8** [comment]
Godot 3.5 still lacks the backports for fixes like https://github.com/godotengine/godot/pull/66490 and followup bug fixes https://github.com/godotengine/godot/pull/66784.

**Comment by akien-mga** [comment]
Fixed by #68988.
